tools-ui/src/main/java/com/hedera/hashgraph/client/ui/SettingsPaneController.java

```diff
@@ -616,24 +633,38 @@ private void checkTransactionFee() {
 		defaultTransactionFee.setText(Utilities.setHBarFormat(transactionFee));
 	}
 
+	private Identifier calcNodeId(final String account) {
+		return Optional.ofNullable(networkChoicebox.getValue())
+				.filter(value -> value instanceof String)
+				.map(value -> (String) value)
+				.map(network -> Identifier.parse(account, network))
+				.orElseGet(() -> Identifier.parse(account));
+	}
+
 	private void checkNodeID() {
-		final var account = nodeIDTextField.getText();
-		final Identifier accountID;
+		final String account = nodeIDTextField.getText();
 		try {
-			accountID = Identifier.parse(account);
-			if (accountID.isValid()) {
-				controller.setDefaultNodeID(accountID.toReadableString());
-				nodeIDTextField.clear();
-				final Identifier defaultNodeID = Identifier.parse(controller.getDefaultNodeID());
-				final var s = defaultNodeID.toNicknameAndChecksum(controller.getAccountsList());
-				nodeIDTextField.setText(s);
-				settingScrollPane.requestFocus();
+			LOG.info("Node ID text field changed to: {}", account);
+			final Identifier calculatedIdentifier = calcNodeId(controller.getDefaultNodeID());
+			final String calculatedAccount = calculatedIdentifier.toNicknameAndChecksum(controller.getAccountsList());
+			if (Objects.equals(calculatedAccount, account)) {
+				controller.setDefaultNodeID(calculatedIdentifier.toReadableString());
```

Comment:
If the calculatedAccount is from the Identifier that was calculated from the defaultNodeId in the controller, why set the defaultNodeId here? I don't see it being an issue, but just wondering if I am missing something, or if this is unneeded.

Reply:
The account is != the nodeID. I had some problems to get this and can not describe the model in detail but as far as I remember I needed to add this